marklogic-client-api/src/main/java/com/marklogic/client/impl/DOMWriter.java

```diff
@@ -136,7 +136,8 @@ public void serializeAttributes(NamedNodeMap attributes) throws XMLStreamExcepti
     }
   }
   public void serializeText(Text text) throws XMLStreamException {
-    serializer.writeCharacters(text.getData());
+    if(text != null && text.getData() != null)
+      serializer.writeCharacters(text.getData());
```

Comment:
Checking whether text is null seems properly defensive.  

If text is not null, I believe that text.getData() is guaranteed not to be null, so that part of the check is unnecessary.